YoCode/FileChangeChecker.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Cryptography;
+
+namespace YoCode
+{
+    public class FileChangeChecker
+    {
+        public static bool ProjectIsModified(IDirectory dir)
+        {
+            if (dir.OriginalPaths.Count != dir.ModifiedPaths.Count)
+            {
+                return true;
+            }
+            else
+            {
+                List<Stream> originalFileStreams = dir.ReturnOriginalPathFileStream();
+                List<Stream> modifiedFileStreams = dir.ReturnModifiedPathFileStream();
+
+                for (int i = 0; i < modifiedFileStreams.Count; i++)
+                {
+                    if (FileIsModified(originalFileStreams[i], modifiedFileStreams[i]))
```

Comment:
Is i-th element in the originalFileStreams always going to be the same file (f.e. Program.cs) as in the modifiedFileStreams. let's say when  a user changes file location but doesn't modify it? Or should that be considered as modifying the file?